game.js

```diff
@@ -454,11 +453,24 @@ class Game {
         const winloss = (this.state.winner === this.conn.bot_id ? "W" : "L");
         this.log(`Game over.   Result: ${col}+${res}  ${winloss}`);
 
-        if (this.bot) {
-            this.bot.gameOver();
-            this.ensureBotKilled();
+        // Notify bot of end of game and send score
+        if (config.farewell_score && this.state) {
+            const sendTheScore = (score) => {
+                if (score) this.log(`Bot thinks the score was ${score}`);
+                if (res !== "R") this.sendChat(`Final score was ${score} according to the bot.`, "discussion");
+                if (this.bot) { // only kill the bot after it processed this
+                    this.bot.gameOver();
+                    this.ensureBotKilled();
+                }
+            };
+            this.bot.command('final_score', sendTheScore, false, true); // allow bot to process end of game
+        }
+        else {
+            if (this.bot) {
+                this.bot.gameOver();
```

Comment:
this code is duplicated

```Js
                if (this.bot) { // only kill the bot after it processed this
                    this.bot.gameOver();
                    this.ensureBotKilled();
                }
```

you may want to add a method (or a function instead) and do 2 calls

Reply:
also the nested `else` then `if` could be written in one `else if` i think, even though it may not be best way to code this

Reply:
fixed